R/content.R

```diff
@@ -198,31 +198,53 @@ method(format, ContentToolRequest) <- function(x, ...) {
 
 #' @rdname Content
 #' @export
-#' @param value,error Either the results of calling the function if
-#'   it succeeded, otherwise the error message, as a string. One of
-#'   `value` and `error` will always be `NULL`.
+#' @param value The results of calling the tool function, if it succeeded.
+#' @param error The error message, as a string, or the error condition thrown
+#'   as a result of a failure when calling the tool function. Must be `NULL`
+#'   when the tool call is successful.
 ContentToolResult <- new_class(
   "ContentToolResult",
   parent = Content,
   properties = list(
     id = prop_string(),
     value = class_any,
-    error = prop_string(allow_null = TRUE)
+    error = new_property(
+      class = NULL | class_character | new_S3_class("condition"),
+      default = NULL,
+      validator = function(value) {
+        if (is.null(value) || inherits(value, "condition")) {
+          return()
+        }
+
+        if (is.character(value) && length(value) == 1) {
```

Comment:
```suggestion
        if (is_string(value) {
```

But I think you could then combine with the previous `if`.